File: e2/java/Solution.java
class Solution {
    /*
     * param n: As desciption
     * return: An integer, denote the number of trailing zeros in n!
     */
    public long trailingZeros(long n) {
        // write your code here
        long zeros = 0L;
        long radix = 5L;
        while(true) {
            long tmp = n / radix;
            if (tmp == 0) {
                break;
            }
            zeros += tmp;
            radix *= 5L;
        }
        return zeros;
    }
};

File: e2/python/solution.py
class Solution:
    # @param n a integer
    # @return ans a integer
    def trailingZeros(self, n):
        zeros = 0
        radix = 5
        while True:
            tmp = n / radix
            if tmp == 0:
                return zeros
            zeros += tmp
            radix *= 5
